helm dependency update panics on symlink

# [SSCCE](http://www.sscce.org)
To reproduce, run:

```bash
helm create foo
helm create bar

echo '
dependencies:
  - name: bar
    version: 0.1.0
' >> foo/Chart.yaml

ln -s $(pwd)/bar foo/charts/bar

helm dependency update foo
```

It will fail with 
```
walk.go:74: found symbolic link in path: /.../foo/charts/bar resolves to /.../bar
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "pmd" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Dependency bar did not declare a repository. Assuming it exists in the charts directory
Deleting outdated charts
Could not verify foo/charts/bar for deletion: cannot load a directory (Skipping)panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x53b980d]

goroutine 1 [running]:
helm.sh/helm/v3/pkg/chart.(*Chart).Name(...)
	helm.sh/helm/v3/pkg/chart/chart.go:76
helm.sh/helm/v3/pkg/downloader.(*Manager).safeMoveDeps(0xc0001a90e0, {0xc000010088, 0x1, 0x57}, {0xc000befc70, 0xd}, {0xc000befc56, 0xa})
	helm.sh/helm/v3/pkg/downloader/manager.go:452 +0x48d
helm.sh/helm/v3/pkg/downloader.(*Manager).downloadAll(0xc0001a90e0, {0xc000010088, 0xc0004d2e20, 0x1})
	helm.sh/helm/v3/pkg/downloader/manager.go:371 +0x176b
helm.sh/helm/v3/pkg/downloader.(*Manager).Update(0xc0001a90e0)
	helm.sh/helm/v3/pkg/downloader/manager.go:201 +0x109
main.newDependencyUpdateCmd.func1(0xc00036c780, {0xc0004f42b0, 0x1, 0x1})
	helm.sh/helm/v3/cmd/helm/dependency_update.go:74 +0x35c
github.com/spf13/cobra.(*Command).execute(0xc00036c780, {0xc0004f4280, 0x1, 0x1})
	github.com/spf13/cobra@v1.2.1/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc00026b900)
	github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.2.1/command.go:902
main.main()
	helm.sh/helm/v3/cmd/helm/helm.go:87 +0x234
```


Output of `helm version`:

```
version.BuildInfo{Version:"v3.7.0", GitCommit:"eeac83883cb4014fe60267ec6373570374ce770b", GitTreeState:"clean", GoVersion:"go1.17"}
```

Hi I am facing a similar issue as well when using `helm package . --dependency-update` to resolve charts from oci (azure). 
I have to use `helm dependency update` first to succeed with `helm package .`. 

```
$ helm version
version.BuildInfo{Version:"v3.7.0", GitCommit:"eeac83883cb4014fe60267ec6373570374ce770b", GitTreeState:"clean", GoVersion:"go1.16.8"}

$ helm package . --dependency-update
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x185a9ba]

goroutine 1 [running]:
helm.sh/helm/v3/internal/experimental/registry.(*Client).Pull(0x0, 0xc000e10d70, 0x43, 0x0, 0x0, 0x0, 0x43, 0xc0012435e0, 0x44f0a5)
	helm.sh/helm/v3/internal/experimental/registry/client.go:235 +0x21a
helm.sh/helm/v3/pkg/getter.(*OCIGetter).get(0xc000d28b40, 0xc000e72700, 0x34, 0xc001243678, 0xc000e72700, 0x34)
	helm.sh/helm/v3/pkg/getter/ocigetter.go:57 +0x23b
helm.sh/helm/v3/pkg/getter.(*OCIGetter).Get(0xc000d28b40, 0xc000e72700, 0x34, 0xc00064b900, 0x7, 0x8, 0xc000d28b40, 0x0, 0x0)
	helm.sh/helm/v3/pkg/getter/ocigetter.go:36 +0x77
helm.sh/helm/v3/pkg/downloader.(*ChartDownloader).DownloadTo(0xc001243a98, 0xc0003a07d0, 0x34, 0xc0003a0805, 0x14, 0xc00082ccf0, 0x2d, 0x8, 0x0, 0x0, ...)
	helm.sh/helm/v3/pkg/downloader/chart_downloader.go:99 +0x142
helm.sh/helm/v3/pkg/downloader.(*Manager).downloadAll(0xc000197290, 0xc000562000, 0x3, 0x3, 0x0, 0x0)
	helm.sh/helm/v3/pkg/downloader/manager.go:360 +0xd95
helm.sh/helm/v3/pkg/downloader.(*Manager).Update(0xc000197290, 0xc000197290, 0x225ac48)
	helm.sh/helm/v3/pkg/downloader/manager.go:201 +0x1b7
main.newPackageCmd.func1(0xc0002a8a00, 0xc000680ca0, 0x1, 0x2, 0x0, 0x0)
	helm.sh/helm/v3/cmd/helm/package.go:99 +0x431
github.com/spf13/cobra.(*Command).execute(0xc0002a8a00, 0xc000680c80, 0x2, 0x2, 0xc0002a8a00, 0xc000680c80)
	github.com/spf13/cobra@v1.2.1/command.go:856 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc00050ec80, 0xc0005b2370, 0x1, 0xc000563f60)
	github.com/spf13/cobra@v1.2.1/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.2.1/command.go:902
main.main()

$ helm dependency update
[...]
Update Complete. ⎈Happy Helming!⎈
Saving 3 charts
Downloading xxx from repo oci://xxx.azurecr.io/helm/xxx/main/
Pulled: xxx.azurecr.io/helm/xx/main/xxx:1000.0.0-preview.154
Digest: sha256:...
Downloading Downloading xxx from repo oci://xxx.azurecr.io/helm/xxx/main/
Pulled: xxx.azurecr.io/helm/xx/main/xxx:1000.0.0-preview.154
Digest: sha256:...
Downloading xxx from repo oci://xxx.azurecr.io/helm/xxx/main/
Pulled: xxx.azurecr.io/helm/xx/main/xxx:1000.0.0-preview.154
Digest: sha256:...
Deleting outdated charts

$ helm package .
Successfully packaged chart and saved it to: ...
```

--- a/pkg/downloader/manager.go
+++ b/pkg/downloader/manager.go
@@ -446,7 +446,8 @@ func (m *Manager) safeMoveDeps(deps []*chart.Dependency, source, dest string) er
 			fname := filepath.Join(dest, file.Name())
 			ch, err := loader.LoadFile(fname)
 			if err != nil {
-				fmt.Fprintf(m.Out, "Could not verify %s for deletion: %s (Skipping)", fname, err)
+				fmt.Fprintf(m.Out, "Could not verify %s for deletion: %s (Skipping)\n", fname, err)
+				continue
 			}
 			// local dependency - skip
 			if isLocalDependency[ch.Name()] {

